localstack/services/sns/provider.py

```diff
@@ -109,173 +96,32 @@ def on_after_init(self):
         register_sns_api_resource(ROUTER)
 
     @staticmethod
-    def get_store(account_id: str = None, region: str = None) -> SnsStore:
-        return sns_stores[account_id or get_aws_account_id()][region or aws_stack.get_region()]
+    def get_store(account_id: str, region_name: str) -> SnsStore:
+        return sns_stores[account_id][region_name]
 
     @staticmethod
-    def _get_moto_backend(context: RequestContext) -> SNSBackend:
-        return sns_backends[context.account_id][context.region]
-
-    def add_permission(
-        self,
-        context: RequestContext,
-        topic_arn: topicARN,
-        label: String,
-        aws_account_id: DelegatesList,
-        action_name: ActionsList,
-    ) -> None:
-        call_moto(context)
-
-    def check_if_phone_number_is_opted_out(
-        self, context: RequestContext, phone_number: PhoneNumber
-    ) -> CheckIfPhoneNumberIsOptedOutResponse:
-        moto_response: CheckIfPhoneNumberIsOptedOutResponse = call_moto(context)
-        return moto_response
-
-    def create_sms_sandbox_phone_number(
-        self,
-        context: RequestContext,
-        phone_number: PhoneNumberString,
-        language_code: LanguageCodeString = None,
-    ) -> CreateSMSSandboxPhoneNumberResult:
-        call_moto(context)
-        return CreateSMSSandboxPhoneNumberResult()
-
-    def delete_sms_sandbox_phone_number(
-        self, context: RequestContext, phone_number: PhoneNumberString
-    ) -> DeleteSMSSandboxPhoneNumberResult:
-        call_moto(context)
-        return DeleteSMSSandboxPhoneNumberResult()
-
-    def get_endpoint_attributes(
-        self, context: RequestContext, endpoint_arn: String
-    ) -> GetEndpointAttributesResponse:
-        moto_response: GetEndpointAttributesResponse = call_moto(context)
-        return moto_response
-
-    def get_platform_application_attributes(
-        self, context: RequestContext, platform_application_arn: String
-    ) -> GetPlatformApplicationAttributesResponse:
-        moto_response = call_moto(context)
-        return select_from_typed_dict(GetPlatformApplicationAttributesResponse, moto_response)
```

Comment:
question: there seems to be a `select_from_typed_dict` in this one specific API operation, not sure if it was AWS validated, but we might leave it just in case?

Reply:
The invocation is redundant. The [response structure returned by moto](https://github.com/localstack/moto/blob/3166d063c2e00cf0230b8c35da31b45e5a4bd568/moto/sns/responses.py#L452-L454) is identical to [botocore definition](https://github.com/localstack/localstack/blob/800c54e29231caf5fcd78d5be52a55cb1fe4f453/localstack/aws/api/sns/__init__.py#L408-L409).